### DISCUSSION TITLE
Set lang attribute at build time

### DISCUSSION THREAD

**Discussion by byte-artisan** [vercel/next.js#16797] [Ideas]
# Feature request
Allow to setup different lang attribute values to different pages, at build time 
Currently it set on document and so is the same for all pages.

## Is your feature request related to a problem? Please describe.
Indicate the language for the page is important for SEO. Chrome lighthouse report the lack of it.
I can't set it up using client code but that is not the best away for all crawlers and it can't be done for amp pages.

## Describe the solution you'd like
Be able declare it in each page. If declared on the page should override the value defined at the document level.

The logical place to place this functionality should be the meta module.

**Comment by teehemkay** [comment]
Just had to find a way to do it. 
In my case the `lang` is one of the properties returned by `getStaticProps`. To use it to set the `<html lang="...">`  attribute, I use the following code snippet in my [`./pages/_map.js`](https://nextjs.org/docs/advanced-features/custom-app) custom App page:

```javascript
import React, { useEffect } from 'react'
// ...

const MyApp = ({ Component, pageProps }) => {
  useEffect(() => {
    document.documentElement.lang = pageProps.lang;
  });

  return <Component {...pageProps} />;
};

export default MyApp;
```

HTH

**Reply by byte-artisan** [reply]
works for most cases, but not on amp pages, since they don't run client code. That is why i requested.
